### PR TITLE
Added support for mailto: links. 

### PR DESCRIPTION
**Pull request by PieterAelse**
EG: "mailto:piotapps@gmail.com" will be a linked "piotapps@gmail.com".

Later TODO: support for `?subject=...&body=...`

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -22,7 +22,7 @@ class LinkifyExample extends StatelessWidget {
                 throw 'Could not launch $url';
               }
             },
-            text: "Made by https://cretezy.com",
+            text: "Made by https://cretezy.com\n\nMail: mailto:charles@cretezy.com",
```
Comment (Cretezy) [inline example/lib/main.dart]: Nice but maybe let's not put my email here, feels a bit too personal!

File: lib/linkify.dart [hunk 2]
```diff
@@ -40,29 +45,47 @@ List<LinkifyElement> linkify(String text, {bool humanize = false}) {
...
-    if (match.group(1).isNotEmpty) {
-      list.add(TextElement(match.group(1)));
+    if (urlMatch.group(1).isNotEmpty) {
+      list.add(TextElement(urlMatch.group(1)));
     }
 
-    if (match.group(2).isNotEmpty) {
+    if (urlMatch.group(2).isNotEmpty) {
       if (humanize ?? false) {
-        print("humanizing ${match.group(2)}");
```
Comment (Cretezy) [inline lib/linkify.dart]: These debug lines shouldn't be there at all. I left them in by accident while debugging one of the past issues

File: lib/linkify.dart [hunk 3]
```diff
@@ -25,11 +25,16 @@ class TextElement extends LinkifyElement {
...
 }
 
-final _linkifyRegex = RegExp(
+final _linkifyUrlRegex = RegExp(
   r"(\n*?.*?\s*?)((?:https?):\/\/[^\s/$.?#].[^\s]*)",
   caseSensitive: false,
 );
 
+final _linkifyEmailRegex = RegExp(
+  r"(\n*?.*?\s*?)((mailto:)[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4})",
```
Comment (Cretezy) [inline lib/linkify.dart]: I would make the mailto: portion optional, and remove it in the `linkify` method to make plain emails work

**Comment by PieterAelse** [comment]
Resolved your comments, one discussion point left (see the TODO)

File: lib/flutter_linkify.dart [hunk 4]
```diff
@@ -17,8 +20,12 @@ class Linkify extends StatelessWidget {
   final TextStyle linkStyle;
 
   /// Callback for tapping a link
-  final LinkCallback onOpen;
+  final LinkCallback onLinkOpen;
```
Comment (Cretezy) [inline lib/flutter_linkify.dart]: Breaking change, will require a major bump.

File: lib/flutter_linkify.dart [hunk 5]
```diff
@@ -90,7 +106,14 @@ TextSpan buildTextSpan({
...
             recognizer: TapGestureRecognizer()
-              ..onTap = () => _onOpen(element.url),
+              ..onTap = () => _onLinkOpen(element.url),
+          );
+        } else if (element is EmailElement) {
+          return TextSpan(
+              text: element.text,
+              style: linkStyle,
+              recognizer: TapGestureRecognizer()
+            ..onTap = () => _onEmailOpen(element.emailAddress),
```
Comment (Cretezy) [inline lib/flutter_linkify.dart]: Seems like the formatting is a bit off here.

File: lib/linkify.dart [hunk 6]
```diff
@@ -13,6 +13,19 @@ class LinkElement extends LinkifyElement {
   }
 }
 
+/// Represents an element containing an email address
+class EmailElement extends LinkifyElement {
+  final String emailAddress;
```
Comment (Cretezy) [inline lib/linkify.dart]: Is there a case where the email address != displayed address?

File: pubspec.yaml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 name: flutter_linkify
-description: Turns text URLs into clickable inline links in text for Flutter.
-version: 1.1.1
+description: Turns text URLs and Mailto links into clickable inline links in text for Flutter.
+version: 1.2.0
```
Comment (Cretezy) [inline pubspec.yaml]:
```suggestion
version: 2.0.0
```

To match the breaking change above.

We could support both onOpen and onOpenLink (and deprecate onOpen). To to you to make the decision!

**Comment by PieterAelse** [comment]
@Cretezy Will have a look later. What about the TODO mentioned at [flutter_linkify.dart#87](https://github.com/Cretezy/flutter_linkify/pull/8/files#diff-d54b1e7b87104c7324d00ebbead0ea17R87 )

File: lib/linkify.dart (hunk 6 above)
Comment (PieterAelse) [reply lib/linkify.dart]: Not now I think.. But it will be when support for `?subject=...&body=...` is added.

Comment (Cretezy) [reply lib/linkify.dart]: I think that's still gonna be the same though, since subject, body, etc is simply gonna be properties (not shown)

**Comment by Cretezy** [comment]
I'll take a look at this later tonight. Looks great thogh!

**Comment by PieterAelse** [comment]
@Cretezy any updates?

**Comment by Cretezy** [comment]
Totally forgot about this, sorry! Will check when I'm home.

**Comment by Cretezy** [comment]
Merged. Will prepare 2.0.0 release, I want to change the API a little.

**Pull request merged**